### PR TITLE
badger: update storage to use badgerV3

### PR DESCRIPTION
**Pull request by Jictyvoo**
Implementing a new version of badger to fix problems with WSL, as discussed in #67 

- [x] Add compatibility with Badger-V3
- [x] Add badger config to storage config options
- [x] Check the corresponding of WithTruncate function

### CODE REVIEW

**Comment by Jictyvoo** [comment]
I still make some changes, as said in description, to give users more freedom

**Comment by ReneWerner87** [comment]
first of all thanks for the contribution

instead of making a new version in the extra folder, what do you think about replacing the old version ?

**Comment by Jictyvoo** [comment]
> first of all thanks for the contribution
> 
> instead of making a new version in the extra folder, what do you think about replacing the old version ?

The problem with replacing the old version is data in v1 being incompatible with v2, which is also incompatible with v3 But if you see no problem, I can replace instead of creating a new folder

**Comment by ReneWerner87** [comment]
yes no problem, we will mention this in the new release, people should update

**Comment by Jictyvoo** [comment]
Done

File: badger/badger.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ func New(config ...Config) *Storage {
 	cfg := configDefault(config...)
 
 	// Set options
-	opt := badger.DefaultOptions(cfg.Database).WithTruncate(true).WithLogger(nil)
+	opt := cfg.BadgerOptions
+	// TODO see if as a new function similar to WithTruncate(true)
```
Comment (ReneWerner87) [inline badger/badger.go:24]: please check this TODO

**Review by ReneWerner87 — APPROVED** [review]

File: badger/badger.go (hunk 1 above)
Comment (Jictyvoo) [reply badger/badger.go:24]: Looks like this option or any similar no more exist

The most similar function that I found is [func (Options) WithCompactL0OnClose](https://pkg.go.dev/github.com/dgraph-io/badger/v3@v3.2011.1#Options.WithCompactL0OnClose)

So, I'll remove the TODO

**Comment by Jictyvoo** [comment]
Any more revisions?

**Pull request merged**